### PR TITLE
DataService enhancement with database indexing

### PR DESCRIPTION
**Pull request by ymai**
Include additional indexing to DataStore to reduce the amount of time for query data for deletion during housekeeping

### CODE REVIEW

**Comment by cdealti** [comment]
@ymai you need to sign the [ECA](https://www.eclipse.org/legal/ECA.php) at Eclipse.

Also make sure to read [how to contribute](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md) (code formatting)

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java [hunk 1]
```diff
@@ -691,6 +696,31 @@ private synchronized void execute(String sql, Timestamp timestamp, Integer... pa
             close(conn);
         }
     }
+    
+    private void createIndex(String indexname, String table, String order)
+			throws KuraStoreException 
+	{
+		// Introduced in 2.0.4, create index for fluke_messages
```
Comment (cdealti) [inline kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java]: Please remove this

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java [hunk 2]
```diff
@@ -182,6 +182,11 @@ private void init(int houseKeeperInterval, int purgeAge, int capacity) throws Ku
                 throw e;
             }
         }
+        
+        // Introduced in 3.0??.  create index that speeds up the process of deleting expired messages during housekeeping task
```
Comment (cdealti) [inline kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java]: Please remove this

**Review by cdealti — CHANGES_REQUESTED** [review]

File: kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java (hunk 2 above)
Comment (cdealti) [reply kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/store/DbDataStore.java]: Please also update the copyright header:

Copyright (c) 2011, 2017 Eurotech and/or its affiliates

**Review by cdealti — CHANGES_REQUESTED** [review]

**Comment by cdealti** [comment]
@ymai have you signed the ECA with the same email used to signoff the commit?

There a strange email address if you click on the failed ip-validation details below:

https://dev.eclipse.org/eclipse-webhook/services/status_details.php?id=58e7b40c63153

**Comment by dwoodard1** [comment]
@ymai @cdealti I am not sure why the check is failing. I have verified using [this lookup tool](https://projects.eclipse.org/user/cla/validate) that the email address used to sign the commit is valid. I think if you update your reviews we can merge.

**Comment by ymai** [comment]
Hi @cdealti, @dwoodard1 ,  It is signed now with my eclipse account.

**Comment by cdealti** [comment]
This has been manually merged after rewording the commit message.

**Pull request closed without merging**